packages/core/src/generator/index.ts

```diff
@@ -177,18 +177,35 @@ export class UnoGenerator<Theme extends object = object> {
       }
 
       const preflightLayerSet = new Set<string>([])
+
       this.config.preflights.forEach(({ layer = LAYER_PREFLIGHTS }) => {
         layerSet.add(layer)
         preflightLayerSet.add(layer)
       })
 
+      const importSet = new Set<string>()
+      const hoistAtImport = (css: string) => {
+        const importRe = /(@import\s+(?:url\(["']?[^"')]+["']?\)\s*;|["'][^"')]+["']\s*;|[^;]+\s*;))/gm
+        const imports = css.match(importRe)
+
+        if (imports != null) {
+          imports.forEach(i => importSet.add(i))
+          return css.replace(importRe, '').replace(/^\s*/gm, '')
+        }
+
+        return css
+      }
+
       preflightsMap = Object.fromEntries(
         await Promise.all(Array.from(preflightLayerSet).map(
           async (layer) => {
             const preflights = await Promise.all(
               this.config.preflights
                 .filter(i => (i.layer || LAYER_PREFLIGHTS) === layer)
-                .map(async i => await i.getCSS(preflightContext)),
+                .map(async (i) => {
+                  const css = await i.getCSS(preflightContext)
+                  return css ? hoistAtImport(css) : undefined
+                }),
```

Comment:
Should put in the rule, not the core, as it, again, coupled and affect every generation

Reply:
You mean, when writing them, put them into the import layer actively?
```ts
prelights: [
 {
   layer: 'imports',
   getCSS: ()=> `@import ....`
 }
]
```

> coupled and affect every generation

Although it is a bit coupled, uno can correct it in the final stage by Regexp, and extract the correct import css to the layer. Make it a default behavior? 😅